src/manager/checkpoint/submit/mod.rs

```diff
@@ -0,0 +1,31 @@
+mod timeout;
+
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: MIT
+use anyhow::Result;
+use async_trait::async_trait;
+use fvm_shared::address::Address;
+use fvm_shared::clock::ChainEpoch;
+use ipc_sdk::subnet_id::SubnetID;
+use serde::Serialize;
+
+/// The submission strategy. Depending on the speed requirement, the strategy to submit the checkpoint
+/// can vary. For example, we can have checkpoint submission without waiting for the message to be
+/// executed, or we can have checkpoint submission that waits for a specific timeout period.
+#[async_trait]
+pub trait SubmissionStrategy {
```

Comment:
I would probably remove this abstraction for now, and have `submit_checkpoint` as part of the `CheckpointManager`, and it checkpoint manager can implement the checkpoint submission strategy at will. I don't expect an agent implementation to require more than one.

Reply:
remove trait already